src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -252,9 +258,131 @@ internal bool CanGenerateLocal()
+                            {
+                                this.OfferReadOnlyFieldFirst = IsReadOnly(previousAssignedSymbol) ||
+                                                               IsReadOnly(nextAssignedSymbol);
+                            }
+
+                            this.AfterThisLocation = this.AfterThisLocation ?? previousAssignedSymbol?.Locations.FirstOrDefault();
+                            this.BeforeThisLocation = this.BeforeThisLocation ?? nextAssignedSymbol?.Locations.FirstOrDefault();
+                        }
+                    }
+                }
+            }
+
+            private ISymbol TryGetAssignedSymbol(
+                SemanticDocument document, SymbolKind symbolKind,
+                ChildSyntaxList children, int index,
+                CancellationToken cancellationToken)
+            {
+                var syntaxFacts = document.Document.GetLanguageService<ISyntaxFactsService>();
+                if (index >= 0 && index < children.Count)
+                {
+                    var sibling = children[index];
+                    if (sibling.IsNode)
+                    {
+                        var siblingNode = sibling.AsNode();
+                        if (syntaxFacts.IsSimpleAssignmentStatement(siblingNode))
+                        {
+                            syntaxFacts.GetPartsOfAssignmentStatement(
+                                siblingNode, out var left, out var right);
+
+                            var symbol = document.SemanticModel.GetSymbolInfo(left, cancellationToken).Symbol;
+                            if (symbol?.Kind == symbolKind &&
+                                symbol.ContainingType.Equals(this.ContainingType))
+                            {
+                                return symbol;
+                            }
+                        }
+                    }
+                }
+
+                return null;
+            }
+
+            private bool IsReadOnly(ISymbol symbol)
+            {
+                if (symbol == null)
+                {
+                    return false;
+                }
+
+                switch (symbol)
+                {
+                    case IFieldSymbol field: return field.IsReadOnly;
+                    case IPropertySymbol property:
+                        if (property.SetMethod == null)
+                        {
+                            return true;
+                        }
+
+                        if (property.ContainingType.GetMembers().OfType<IFieldSymbol>().Any(
+                                f => property.SetMethod.Equals(f.AssociatedSymbol)))
+                        {
+                            return true;
+                        }
+
+                        return false;
+                }
+
+                throw ExceptionUtilities.Unreachable;
+            }
+
+            private int GetStatementindex(ChildSyntaxList children, SyntaxNode statement)
```

Comment:
I'm thinking about some of the multi-thousand line `InitializeComponent` methods I've seen.